backend/src/v5/middleware/dataConverter/inputs/teamspaces/projects/models/commons/tickets.comments.js

```diff
@@ -16,29 +16,39 @@
  */
 
 const { createResponseCode, templates } = require('../../../../../../../utils/responseCodes');
+const { types, utils: { stripWhen } } = require('../../../../../../../utils/helper/yup');
 const { UUIDToString } = require('../../../../../../../utils/helper/uuids');
 const Yup = require('yup');
 const { getCommentById } = require('../../../../../../../models/tickets.comments');
 const { getTicketById } = require('../../../../../../../models/tickets');
 const { getUserFromSession } = require('../../../../../../../utils/sessions');
+const { isEqual } = require('../../../../../../../utils/helper/objects');
 const { isUUIDString } = require('../../../../../../../utils/helper/typeCheck');
 const { respond } = require('../../../../../../../utils/responder');
-const { types } = require('../../../../../../../utils/helper/yup');
 const { validateMany } = require('../../../../../../common');
 
 const CommentsMiddleware = {};
 
 const validateComment = (isNewComment) => async (req, res, next) => {
 	try {
 		const schema = Yup.object().shape({
-			comment: types.strings.longDescription,
+			comment: isNewComment
+				? types.strings.longDescription
+				: stripWhen(types.strings.longDescription, (t) => isEqual(t, req.commentData.comment)),
 			images: Yup.array().min(1).of(
 				isNewComment
 					? types.embeddedImage()
 					: types.embeddedImageOrRef()
 						.test('Image ref test', 'One or more image refs do not correspond to a current comment image ref',
-							(value, { originalValue }) => !isUUIDString(originalValue)
-								|| req.commentData?.images.map(UUIDToString).includes(originalValue)),
+							(value, { originalValue }) => {
+								if (!isUUIDString(originalValue)) return true;
+
+								const historyImages = req.commentData.history?.flatMap((h) => h.images)
```

Comment:
2 things:

1. unless `image` is a variable that's already used, just deconstruct the object in the function declaration
2. if you return a [] instead of undefined then you don't need to do another O(n) loop to filter.
![image](https://user-images.githubusercontent.com/11945337/214589831-c1c58f8c-bc7c-4251-b348-a52cd48a2a21.png)


```suggestion
								const historyImages = req.commentData.history?.flatMap(({images}) => images??[])
```

Reply:
It's probably better to calculate the acceptable refs outside of this test, as this test is triggered for every element, but the acceptable refs doesn't change